tests/asgi/tests.py

```diff
@@ -237,6 +237,30 @@ async def test_disconnect(self):
         with self.assertRaises(asyncio.TimeoutError):
             await communicator.receive_output()
 
+    async def test_disconnect_both_return(self):
```

Comment:
`test_disconnect_both_return()` and `test_streaming()` work without this patch. Extra test coverage is welcome but we should move it to a separate commit/PR.